### PR TITLE
feat: delete local files when offline, refactor delete process

### LINKED ISSUE #8: Delete appInstanceResources when offline

### PR DESCRIPTION
**Pull request by pyphilia**
This PR enables to delete files and related appInstanceResources from a local server (graasp-desktop). The process goes as for uploading files: the appInstanceResource is first removed, then is the file.

closes #8

### CODE REVIEW

File: src/actions/listener.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    switch (type) {
+      case GET_APP_INSTANCE_RESOURCES_SUCCEEDED:
+      case GET_APP_INSTANCE_SUCCEEDED:
+      case PATCH_APP_INSTANCE_RESOURCE_SUCCEEDED:
+      case POST_APP_INSTANCE_RESOURCE_SUCCEEDED:
+        return dispatch({
+          type,
+          payload,
+        });
+      case DELETE_APP_INSTANCE_RESOURCE_SUCCEEDED: {
```
Comment (juancarlosfarah) [inline src/actions/listener.js:39]: Why are you wrapping these two cases in `{}`?

File: src/actions/listener.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      case POST_FILE_FAILED: {
+        // the error message may be passed in payload
+        const errorMessage = _.isString(payload)
+          ? payload
+          : FILE_UPLOAD_FAILED_MESSAGE;
+        return showWarningToast(errorMessage);
+      }
+      case DELETE_FILE_SUCCEEDED:
+      case DELETE_FILE_FAILED:
+        // do nothing: on faile, the file will still be on the server
```
Comment (juancarlosfarah) [inline src/actions/listener.js:83]: `failed`

File: src/actions/listener.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  const { data } = event;
+  try {
+    const message = JSON.parse(data);
+
+    const { type, payload } = message;
+
+    switch (type) {
+      case POST_FILE_SUCCEEDED:
+        return dispatch(postAppInstanceResource(payload));
+      case POST_FILE_FAILED: {
```
Comment (juancarlosfarah) [inline src/actions/listener.js:74]: Why `{}`? (Same as above).

File: src/components/modes/teacher/TeacherView.js [hunk 4]
```diff
@@ -97,26 +97,27 @@ export class TeacherView extends Component {
     dispatchGetUsers();
   }
 
-  handleDelete = async ({ id, uri }) => {
-    const {
-      dispatchDeleteAppInstanceResource,
-      dispatchDeleteFile,
-    } = this.props;
+  handleDelete = async ({ id, _id, uri }) => {
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/TeacherView.js:100]: From what I see below, this should not need to receive an `_id` so no need to handle that case. (See below).

**Review by juancarlosfarah — CHANGES_REQUESTED** [review]
A complex change, but it looks pretty good. Do you think that we could write a test for this? In any case, here are some minor comments. Otherwise it's looking nice!

File: src/actions/listener.js (hunk 3 above)
Comment (pyphilia) [reply src/actions/listener.js:74]: This one is necessary because a `const` is defined inside.

Comment (juancarlosfarah) [reply src/actions/listener.js:74]: OK. Gotcha.

**Pull request merged**